src/server/gc_worker.rs

```diff
@@ -212,6 +216,53 @@ impl<E: Engine> GcRunner<E> {
         .map_err(Error::from)
     }
 
+    /// Check need gc without getting snapshot.
+    /// If this is not supported or any error happens, returns true to do further check after
+    /// getting snapshot.
+    fn need_gc(&self, ctx: &Context, safe_point: TimeStamp) -> bool {
+        let region_info_accessor = match &self.region_info_accessor {
+            Some(r) => r,
+            None => return true,
+        };
+
+        let db = match &self.local_storage {
+            Some(db) => db,
+            None => return true,
+        };
+
+        let (tx, rx) = mpsc::channel();
+        if region_info_accessor
+            .find_region_by_id(
```

Comment:
How about getting region range in ` AutoGcConfig::gc_next_region()`? If so, the `GcWorker` needn't contain the `RegionInfoAccessor`.

Reply:
`AutoGcConfig` is used by `GCManager` and `GCRunner` cannot access it.

Reply:
Yes. Carry key range in `GcTask::Gc`.

Reply:
But it's still possible that TiDB or other clients sends GC requests to TiKV, which cannot carry the range. Or is it ok not to do the check in that range?

Reply:
O, I forgot it. It's needed to get key range in `GcWorker`.